src/methods/geom_relations/geom_geom_processors.jl

```diff
@@ -608,24 +606,16 @@ function _line_filled_curve_interactions(
                         so calculate if segment endpoints and intersections are
                         in/out of filled curve
                         =#
-                        ipoints = intersection_points(
-                            GI.Line([l_start, l_end]),
-                            curve
-                        )
+                        ipoints = intersection_points(GI.Line(StaticArrays.SVector(l_start, l_end)), curve)
                         npoints = length(ipoints)  # since hinge, at least one
                         dist_from_lstart = let l_start = l_start
                             x -> _euclid_distance(Float64, x, l_start)
                         end
                         sort!(ipoints, by = dist_from_lstart)
                         p_start = _tuple_point(l_start)
                         for i in 1:(npoints + 1)
-                            p_end = i ≤ npoints ?
-                                _tuple_point(ipoints[i]) :
-                                l_end
-                            mid_val = _point_filled_curve_orientation(
-                                (p_start .+ p_end) ./ 2,
-                                curve,
-                            )
+                            p_end = i ≤ npoints ? _tuple_point(ipoints[i]) : l_end
+                            mid_val = _point_filled_curve_orientation((p_start .+ p_end) ./ 2, curve; exact)
```

Comment:
You would have to perform that division within the predicate, and add a grouping statement so that the error checking can handle it!

Reply:
I think I can actually get rid of the checks when it is checking the midpoint of a shared edge by re-writing the logic a bit! We still might need the increased precision for other cases, but that is the one I see the greatest issues with.